CrossDistroLDAPInfra/main.tf

```diff
@@ -0,0 +1,115 @@
+# Input Variables
+variable "access_key" {
+  description = "AWS access key"
+  type        = string
+}
+
+variable "secret_key" {
+  description = "AWS secret key"
+  type        = string
+}
+
+# Provider Configuration
+provider "aws" {
+  region     = "us-east-1"
+  access_key = var.access_key
+  secret_key = var.secret_key
+}
+
+# Common Security Group
+resource "aws_security_group" "allow_ssh" {
+  name_prefix = "allow_ssh"
+
+  ingress {
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+# Debian Instance
+resource "aws_instance" "debian" {
+  ami             = "ami-064519b8c76274859"
```

Comment:
https://www.itwonderlab.com/aws-terraform-tutorial-aws-ami/

you should not directly hardcode AMI identifiers (like the AMI ID) in your Terraform file; instead, use a Terraform data source (data "aws_ami") to dynamically fetch the correct AMI ID based on filters like name, owner, and other relevant criteria, ensuring your configuration is more flexible and less prone to errors if the AMI ID changes.